src/Evolution/DgSubcell/Actions/TciAndRollback.hpp

```diff
@@ -130,9 +130,25 @@ struct TciAndRollback {
           return false;
         }());
 
-    const bool subcell_allowed_in_block = not std::binary_search(
-        subcell_options.only_dg_block_ids().begin(),
-        subcell_options.only_dg_block_ids().end(), element.id().block_id());
+    // Loop over block neighbors and if neighbor id is inside of
+    // subcell_options.only_dg_block_ids(), then bordering DG-only block
+    const bool bordering_DG_block = alg::any_of(
+        element.neighbors(),
+        [&subcell_options](const auto& direction_and_neighbor) {
+          const bool first_block_id =
+              direction_and_neighbor.second.ids().begin()->block_id();
+          return alg::found(subcell_options.only_dg_block_ids(),
```

Comment:
I think this could be a `binary_search` instead of `found` since the `only_dg_block_ids()` is sorted